iati/tests/test_resources.py

```diff
@@ -50,193 +177,492 @@ class TestResourceFolders(object):
 
-    def test_schema_activity_string(self, std_ver_minor_inst_valid_known):
-        """Check that the Activity schema file contains content.
+        assert len(result) == 1
+        assert result[0] == iati.resources.create_schema_path(func_and_name.schema_name, std_ver_minor_mixedinst_valid_known)
 
-        Todo:
-            Change fixture to mixedinst.
+    def test_get_schema_paths_minor_unknown(self, std_ver_all_mixedinst_valid_unknown, schema_path_func_all):
+        """Test getting a list of Org or Activity Schema paths. The requested version is not known by pyIATI."""
+        result = schema_path_func_all(std_ver_all_mixedinst_valid_unknown)
 
-        """
-        path = iati.resources.create_schema_path('iati-activities-schema', std_ver_minor_inst_valid_known)
+        assert result == []
 
-        content = iati.utilities.load_as_string(path)
+    def test_get_schema_paths_independent(self, schema_path_func_all):
+        """Test getting a list of version-independent Org or Activity Schemas."""
+        result = schema_path_func_all(iati.version.STANDARD_VERSION_ANY)
+
+        assert result == []
+
+    def test_get_schema_paths_major_known(self, std_ver_major_uninst_valid_known, func_and_name):
+        """Test getting a list of Org or Activity Schema paths. The requested version is a known integer version. The list should contain paths for each supported minor within the major."""
+        versions_at_major = [version for version in iati.version.versions_for_integer(std_ver_major_uninst_valid_known)]
+        expected_path_count = len(versions_at_major)
+
+        result = func_and_name.func(std_ver_major_uninst_valid_known)
+
+        assert len(result) == expected_path_count
+        for version in versions_at_major:
+            assert iati.resources.create_schema_path(func_and_name.schema_name, version) in result
 
-        assert len(content) > 50000
+    def test_get_all_schema_paths_minor_known(self, std_ver_minor_mixedinst_valid_known):
+        """Test getting a list of all Schema paths. The requested version is known by pyIATI."""
+        activity_path = iati.resources.get_activity_schema_paths(std_ver_minor_mixedinst_valid_known)[0]
+        org_path = iati.resources.get_organisation_schema_paths(std_ver_minor_mixedinst_valid_known)[0]
 
-    def test_schema_activity_tree(self, std_ver_minor_inst_valid_known):
-        """Check that the Activity schema loads into an XML Tree.
+        result = iati.resources.get_all_schema_paths(std_ver_minor_mixedinst_valid_known)
 
-        This additionally involves checking that imported schemas also work.
+        assert len(result) == 2
+        assert activity_path in result
+        assert org_path in result
 
-        Todo:
-            Change fixture to mixedinst.
+        # ensure the paths have at least a minimum amount of content in the files they reference
+        for path in result:
+            assert os.path.getsize(path) > 10000
```

Comment:
There should be some checks (here and elsewhere) that the `get_x_paths()` functions return lists of paths that refer to actual files (and that this check is performed should be noted).

Race conditions still need considering, though it provides far stronger guarantees that the `create_x_path()` functions.